recipes/edyn/all/conanfile.py

```diff
@@ -24,80 +25,71 @@ class EdynConan(ConanFile):
-        return "build_subfolder"
-
-    def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
-
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
-
-    def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
 
     def requirements(self):
-        self.requires("entt/3.9.0")
+        if scm.Version(self.version) < "1.2.0":
+            self.requires("entt/3.9.0")
+        else:
+            self.requires("entt/3.10.1")
 
     @property
     def _compiler_required(self):
         return {
             "gcc": "9.3", # GCC 9.3 started supporting attributes in constructor arguments
+            "clang": "8",
+            "apple-clang": "10",
+            "Visual Studio": "16",
+            "msvc": "192"
         }
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 17)
+            build.check_min_cppstd(self, 17)
         try:
             minimum_required_compiler_version = self._compiler_required[str(self.settings.compiler)]
-            if tools.Version(self.settings.compiler.version) < minimum_required_compiler_version:
+            if scm.Version(self.settings.compiler.version) < minimum_required_compiler_version:
                 raise ConanInvalidConfiguration("This package requires C++17 support. The current compiler does not support it.")
         except KeyError:
             self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
-   
+
+    def layout(self):
+        layout.cmake_layout(self, src_folder="edyn")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        files.get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["EDYN_INSTALL"] = True
+        tc.variables["EDYN_BUILD_EXAMPLES"] = False
+        if self.options.floating_type == "double":
+            tc.variables["EDYN_CONFIG_DOUBLE"] = True
+        tc.generate()
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["EDYN_INSTALL"] = True
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
```

Comment:
what happend to the patches?

Reply:
I was under the impression that if there are no patches, that can be omitted? The previous (1.0.0 - https://github.com/conan-io/conan-center-index/pull/10194) did not actually have any patches even though it did run _patch_sources.
Are there "built-in"/CI-wide patches that have to be retrieved and applied?

Reply:
No no, I only see the diff when I review but if there are no actually patches then you are completely correct... I just assumed it was correct the first time 🙈